CMS-Enterprise/sbom-harbor
# add docker-compose configuration; update config loading code

## Summary

This PR introduces a `docker-compose` feature to bootstrap a local, observable development environment.

### Added

- Introduces a `docker-compose` configuration in the `devenv` folder for running an observable harbor stack for local development.
- Adds a `db_connection` method to make use of the `DB_CONNECTION` environment variable. Defaults to the `docker-compose` instance if not set.

### Changed

- Updates the README to include developer machine dependencies and an environment variables section.

## How to test

- Check out the branch
- cd to `devenv` and run `docker compose up`
- cd back to the root directory and run `cargo test`
